src/coffee/import.coffee

```diff
@@ -59,31 +60,35 @@ class Import
           # - process products in batches!!
           # - for each chunk match products -> createOrUpdate
           # - provide a way to accumulate partial results, or just log them to console
-          products = []
           console.log "Mapping #{_.size rawProducts} product(s) ..."
-          for rawProduct in rawProducts
-            products.push @validator.map.mapProduct(rawProduct)
-
-          if _.size(@validator.map.errors) isnt 0
-            Promise.reject @validator.map.errors
-          else
-            console.log "Mapping done. About to process existing product(s) ..."
-            @client.productProjections.staged().all().fetch()
-            .then (payload) =>
-              existingProducts = payload.body.results
-              console.log "Comparing against #{payload.body.total} existing product(s) ..."
-              @initMatcher existingProducts
-              productsToUpdate =
-              if @validator.updateVariantsOnly
-                @mapVariantsBasedOnSKUs existingProducts, products
-              else
-                products
-              console.log "Processing #{_.size productsToUpdate} product(s) ..."
-              @createOrUpdate(productsToUpdate, @validator.types)
-            .then (result) ->
-              # TODO: resolve with a summary of the import
-              console.log "Finished processing #{_.size result} product(s)"
-              Promise.resolve result
+          # for rawProduct in rawProducts
+          #   products.push @validator.map.mapProduct(rawProduct)
+          products = rawProducts.map((p) => @validator.map.mapProduct p)
+          Promise.all(chunk(products, 100).map((ps) => @processProducts(ps)))
+          .then((results) => results.reduce((agg, r) => agg.concat(r) []))
+
+  processProducts: (products) ->
+    if _.size(@validator.map.errors) isnt 0
+      Promise.reject @validator.map.errors
+    else
+      console.log "Mapping done. About to process existing product(s) ..."
+      ids = products.map((p) -> "\"#{p.product.id}\"").join(',')
```

Comment:
Yes, we talked about it and decided to allow different matching to be selectable via command line.
id, sku, slug and custom attribute. But yes sku is the most common case.